### ISSUE TITLE
[2022] Geometric Considerations for Normalization Layers in Equivariant Neural Networks

### ISSUE THREAD

**Issue by mjrs33** [mjrs33/paper#62]
## 概要
分子の物性予測タスクにおいてGNNでの正規化層の重要性を調査した論文。通常のバッチ正規化はGNNではうまく機能しない場合があり、タスク特有の正規化層が開発され、性能を向上させている。今後はユニバーサルな正規化テクニックを開発すべきと提言。

### 論文リンク
https://openreview.net/forum?id=p9fKD1sFog8

### 著者/所属機関
Max Shirokawa Aalto and Ekdeep Singh Lubana and Hidenori Tanaka

### 投稿日付
2022/11/29

## 背景
GNNで分子の特性予測タスクを解く際、BatchNormはうまくいかないことがあるのが実験的に分かっている。これを解決するため独自の正規化テクニックを導入するが、タスク特有のものであり、他のタスクへの適用性は考えられていない。

## 新規性
- 最新のequivariantなモデルに関して、既存の正規化テクニックをまとめた。

- 正規化技術はタスクと同じ幾何学的制約に従うことが必要（ただし十分ではない）であることを示した。

### BatchNormが機能しないケース
ケース1：
GemNetの著者らは論文中で、バッチ正規化は別々の分子間の相関を導入してしまうと述べている。伝統的な正規化手法を導入すると精度が低下したことも観測している。

ケース2：
“3D-Rotation-Equivariant Quaternion Neural Networks”では、回転同変性が破壊されており、改良したBatchNormが使用されている。

ケース3：
“Rotation equivariant vector field networks”では、ベクトルのスケールを揃えるだけで良いと述べられている。角度は上下限があり、既に正規化されているので意味がないうえ、その分布を変えると配向の重要な情報に変更を与えることになってしまう。この論文でもBatchNormを改良したものが用いられている。

ケース4：
SetNormの著者らも、BatchNormにより精度が悪化するうえ、入力間の（本来無い）相関を導入してしまうことを述べている。

いずれのケースも幾何的な制約があるタスクであり、通常のBatchNormを適用すると、その制約を侵害することになり、性能が低下する。

### 正規化層のまとめ
自然言語および画像分野で用いられる正規化層と、GNNで用いられる正規化層をまとめたのが以下。

![1125E09A-0CEB-4E71-9B2C-5F58A163228B](https://user-images.githubusercontent.com/40561031/211120888-2e2f384e-36d0-46a6-b524-4cf164129c54.jpeg)

バッチは $b × x × v$ の形状（バッチサイズ、ノードサイズ、特徴ベクトルサイズ）をしている。 $g$ は特徴量のサブセット、 $|X|$ はデータセット全体のグラフサイズ、 $N$ は近接ノード数の平均。

GNNでは回転同変になるように設計されているものが多く見られる（スケールだけのものなど）。
SchNetとAllegroではエッジの情報を用いて正規化している。SchNetはグラフ内の情報のみだが、Allegroではデータセット全体の情報を使用している（表記合ってる？）。

## 結果
Allegro, GemNet, SchNetに対して、それぞれの論文で提案された正規化層の有無による精度の変化について検証した。

Allegroの結果は以下。

![B23AB21C-A083-4042-8CFC-94BA39F9660D](https://user-images.githubusercontent.com/40561031/211123065-cb44f3d3-e10b-4179-8ca6-d057d43a885f.jpeg)

正規化層無しでは勾配爆発が起こり、正規化層ありの場合と比較して圧倒的にlossが大きかった。

SchNetとGemNetの結果は以下。

![CEDBC610-5742-4B35-B30E-08CC32F7068F](https://user-images.githubusercontent.com/40561031/211123318-4eaa59e8-d8d8-4e31-8c70-0604e6f0f6f2.jpeg)

Allegroほどではないが、正規化層により精度が改善していることがわかる。
